### ISSUE TITLE
Uncaught Error: EEXIST: file already exists, mkdir '~/workspace/vbx-remote/plugins/citytwig/hooks'

### ISSUE THREAD

**Issue by jlorie** [icetee/remote-ftp#570]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.12.2
**Electron Version**: 1.3.6
**System**: Mac OS X 10.12.1
**Thrown From**: [Remote-FTP](https://github.com/mgrenier/remote-ftp) package, v0.9.4

### Stack Trace

Uncaught Error: EEXIST: file already exists, mkdir '/Users/jlorie/workspace/vbx-remote/plugins/citytwig/hooks'

```
At /Users/jlorie/.atom/packages/Remote-FTP/node_modules/mkdirp/index.js:76

Error: EEXIST: file already exists, mkdir '/Users/jlorie/workspace/vbx-remote/plugins/citytwig/hooks'
    at Error (native)
    at Object.fs.mkdirSync (fs.js:922:18)
    at Function.sync (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/mkdirp/index.js:55:12)
    at Object.makeTreeSync (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/fs-plus/lib/fs-plus.js:381:23)
    at /Users/jlorie/.atom/packages/Remote-FTP/lib/connectors/ftp.js:178:18
    at /Users/jlorie/.atom/packages/Remote-FTP/lib/connectors/ftp.js:209:18
    at Object.cb (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/ftp/lib/connection.js:348:5)
    at Parser.<anonymous> (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/ftp/lib/connection.js:117:20)
    at emitTwo (events.js:106:13)
    at Parser.emit (events.js:191:7)
    at Parser._write (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/ftp/lib/parser.js:59:10)
    at doWrite (_stream_writable.js:307:12)
    at writeOrBuffer (_stream_writable.js:293:5)
    at Parser.Writable.write (_stream_writable.js:220:11)
    at Socket.ondata (/Users/jlorie/.atom/packages/Remote-FTP/node_modules/ftp/lib/connection.js:273:20)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:177:18)
    at Socket.Readable.push (_stream_readable.js:135:10)
    at TCP.onread (net.js:542:20)
```

### Commands

```
     -0:25.2.0 application:add-project-folder (atom-text-editor.editor.is-focused)
     -0:09.6.0 remote-ftp:toggle (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "telemetryConsent": "limited"
  }
}
```

### Installed Packages

```coffee
# User
atom-autocomplete-php, v0.20.2 (active)
atom-beautify, v0.29.13 (inactive)
autocomplete-modules, v1.6.7 (active)
highlight-selected, v0.11.2 (active)
hyperclick, v0.0.39 (active)
hyperclick-php, v0.2.1 (active)
js-refactor, v0.7.3 (active)
language-vue, v0.20.0 (active)
linter, v1.11.18 (active)
linter-eslint, v8.0.0 (active)
linter-php, v1.3.1 (active)
minimap, v4.25.6 (active)
minimap-highlight-selected, v4.4.0 (active)
php-cs-fixer, v2.5.9 (active)
refactor, v0.10.0 (active)
Remote-FTP, v0.9.4 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (active)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (active)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.13.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.4 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.2 (active)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (active)
metrics, v1.1.1 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.4 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.2 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.210.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.0 (active)
language-csharp, v0.12.1 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.46.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by ChrisBarberRiley** [comment]
Same problem here.

Uncaught Error: EEXIST: file already exists, mkdir '/Users/chrisbarber-riley/Desktop/Development/domain/httpdocs/system/user/addons/hub_selector'

/Users/chrisbarber-riley/.atom/packages/Remote-FTP/node_modules/mkdirp/index.js:76

**Comment by ngr-t** [comment]
I got a similar error (on Windows 7, Atom 1.14.3 ia32, Remote-FTP 0.10.5).
I could reproduce the error by trying to download a file whose path includes a directory whose name matches a file name of regular file on the local.
e.g. to download remote `path/to/file` but the mirrored local directory has a regular file `path`. Removed the local file and the download became successful.

I can't figure out why there was the file, but maybe just a trivial mistake to choose the `Add File` menu item for `Add Folder`, in my case.

Thanks,
